fastcrypto/src/bls12381/mod.rs

```diff
@@ -272,17 +272,18 @@ impl VerifyingKey for BLS12381PublicKey {
     }
 }
 
-fn get_128bit_scalar<Rng: AllowedRng>(rng: &mut Rng) -> blst_scalar {
-    debug_assert!(BLS_BATCH_RANDOM_SCALAR_LENGTH <= 128);
+fn get_random_scalar<Rng: AllowedRng>(rng: &mut Rng) -> blst_scalar {
+    static_assertions::const_assert!(64 <= BLS_BATCH_RANDOM_SCALAR_LENGTH);
+    static_assertions::const_assert!(BLS_BATCH_RANDOM_SCALAR_LENGTH <= 128);
+
     let mut vals = [0u64; 4];
     loop {
         vals[0] = rng.next_u64();
         vals[1] = rng.next_u64();
 
-        // Check this ^^
-
         // Reject zero as it is used for multiplication.
-        if vals[0] | vals[1] != 0 {
+        let vals1_lsb = vals[1] & ((1 << (BLS_BATCH_RANDOM_SCALAR_LENGTH - 64)) - 1);
```

Comment:
I guess in practice we don't even need to sample `vals[1]` in the first place when requesting for 64bits. That would need two if statements (one for sampling one for checking values are zero), but I'm not sure which is one is better for readability. I'd personally do the two `if`.